inst/include/dust/interface.hpp

```diff
@@ -100,15 +102,88 @@ void dust_set_index(SEXP ptr, cpp11::sexp r_index) {
 }
 
 template <typename T>
-void dust_set_state(SEXP ptr, SEXP r_state, SEXP r_step, bool deterministic) {
-  typedef typename T::real_t real_t;
+cpp11::sexp dust_update_state_set_pars(Dust<T> *obj, cpp11::list r_pars,
+                                       bool set_initial_state) {
+  cpp11::sexp ret = R_NilValue;
+  if (obj->n_pars() == 0) {
+    dust::pars_t<T> pars = dust_pars<T>(r_pars);
+    obj->set_pars(pars, set_initial_state);
+    ret = dust_info<T>(pars);
+  } else {
+    dust::interface::check_pars_multi(r_pars, obj->shape(),
+                                      obj->pars_are_shared());
+    std::vector<dust::pars_t<T>> pars;
+    pars.reserve(obj->n_pars());
+    cpp11::writable::list info_list =
+      cpp11::writable::list(r_pars.size());
+    for (int i = 0; i < r_pars.size(); ++i) {
+      pars.push_back(dust_pars<T>(r_pars[i]));
+      info_list[i] = dust_info<T>(pars[i]);
+    }
+    obj->set_pars(pars, set_initial_state);
+    ret = info_list;
+  }
+  return ret;
+}
+
+template <typename T>
+cpp11::sexp dust_update_state_set(Dust<T> *obj, SEXP r_pars,
+                                  const std::vector<typename T::real_t>& state,
+                                  const std::vector<size_t>& step,
+                                  bool set_initial_state) {
```

Comment:
```suggestion
// There are many components of state (not including rng state which
// we treat separately), we set components always in the order (1)
// pars, (2) state, (3) step
template <typename T>
cpp11::sexp dust_update_state_set(Dust<T> *obj, SEXP r_pars,
                                  const std::vector<typename T::real_t>& state,
                                  const std::vector<size_t>& step,
                                  bool set_initial_state) {
```

Reply:
made this change but moved rather than added the comment